docs/topics/testing/tools.txt

```diff
@@ -1552,6 +1552,29 @@ your test suite.
 
     Output in case of error can be customized with the ``msg`` argument.
 
+.. method:: SimpleTestCase.assertFormValid(form, msg=None)
+
+    .. versionadded:: 1.10
+
+    Asserts that a form is valid. If form is not valid, displays clean output
+    with form errors.
+
+    ``form`` is the form to be validated.
+
+.. method:: SimpleTestCase.assertFormNotValid(form, expected_errors=None, msg=None)
+
+    .. versionadded:: 1.10
+
+   Asserts that a form is not valid and that errors are ``expected_errors``.
+
+   ``form`` is the form to be validated.
+
+   ``expected_errors`` is a list of (field, code) errors that are expected on
+   the form.
```

Comment:
I think the format of `expected_errors` could be better, and more clearly explained. It might be a nice feature to allow any error on a given field rather than having to know the correct code. In particular I think the output for an error raised on the correct field but the wrong error code might have confusing output at the moment.

It would be lovely to have `expected_errors=['name', ('email', 'required')]` as a valid format.

I don't want to derail this as I think the current code is still very useful (and infinitely better than `assertTrue(form.is_valid())`. I do think we should test what happens though if the field with an error is right but the code is wrong.